types/math/math.go

```diff
@@ -110,3 +110,37 @@ func SafeSub(res, x, y *apd.Decimal) error {
 
 	return nil
 }
+
+// SubBalance subtracts the value of y from x and returns the result with arbitrary precision.
+// Returns with ErrInsufficientFunds error if the result is negative.
+func SubBalance(x Dec, y Dec) (Dec, error) {
```

Comment:
```suggestion
func SafeSubBalance(x Dec, y Dec) (Dec, error) {
```

Reply:
Do we need `Balance` suffix? It doesn't need to be related to a balance. I think `SafeSub` or `CheckedSub` is good enough.